src/chrome/chromeDebugAdapter.ts

```diff
@@ -646,75 +655,88 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
+                        await this.sendInitializedEvent();
+                    });
+                }
 
-        this._scriptsById.set(script.scriptId, script);
-        this._scriptsByUrl.set(this.fixPathCasing(script.url), script);
+                if (this._earlyScripts) {
+                    this._earlyScripts.push(script);
+                } else {
+                    await this.sendLoadedSourceEvent(script);
+                }
+            });
 
-        const resolvePendingBPs = (source: string) => {
-            source = source && this.fixPathCasing(source);
-            const pendingBP = this._pendingBreakpointsByUrl.get(source);
-            if (pendingBP && !pendingBP.bpsSet) {
-                this.resolvePendingBreakpoint(pendingBP)
-                    .then(() => this._pendingBreakpointsByUrl.delete(source));
+            if (script.url) {
+                script.url = utils.fixDriveLetter(script.url);
+            } else {
+                script.url = ChromeDebugAdapter.EVAL_NAME_PREFIX + script.scriptId;
             }
-        };
 
-        const mappedUrl = await this._pathTransformer.scriptParsed(script.url);
+            this._scriptsById.set(script.scriptId, script);
+            this._scriptsByUrl.set(this.fixPathCasing(script.url), script);
 
-        const sourceMapsP = this._sourceMapTransformer.scriptParsed(mappedUrl, script.sourceMapURL).then(sources => {
-            if (this._hasTerminated) {
-                return undefined;
-            }
+            const mappedUrl = await this._pathTransformer.scriptParsed(script.url);
 
-            if (sources) {
-                // If break on load is active, check whether we should call resolvePendingBPs
-                if (this.breakOnLoadActive) {
-                    sources
-                        .filter(source => source !== mappedUrl && this._breakOnLoadHelper.shouldResolvePendingBPs(source)) // Tools like babel-register will produce sources with the same path as the generated script
-                        .forEach(resolvePendingBPs);
+            const resolvePendingBPs = (source: string) => {
+                source = source && this.fixPathCasing(source);
+                const pendingBP = this._pendingBreakpointsByUrl.get(source);
+                if (pendingBP && !pendingBP.bpsSet) {
+                    this.resolvePendingBreakpoint(pendingBP)
+                        .then(() => {
+                            this._pendingBreakpointsByUrl.delete(source);
+                            breakpointsAreResolvedDefer.resolve();
+                        });
                 } else {
+                    breakpointsAreResolvedDefer.resolve();
+                }
+            };
+
+            const sourceMapsP = this._sourceMapTransformer.scriptParsed(mappedUrl, script.sourceMapURL).then(async sources => {
+                if (this._hasTerminated) {
+                    return undefined;
+                }
+
+                if (sources) {
```

Comment:
When is this undefined?

Reply:
This code was added for this: https://github.com/Microsoft/vscode-chrome-debug-core/issues/106 in https://github.com/Microsoft/vscode-chrome-debug-core/commit/1bf19e826b389f4033faa76287921600cf03ca16 so I'm assuming that the scenario is:
"sourcemaps can't be resolved ahead of time, or don't exist on disk."